src/server.js

```diff
@@ -0,0 +1,98 @@
+var fs = require('fs'),
+  dns = require('native-dns'),
+  dnsServer = dns.createServer(),
+  https = require('https'),
+  records = {
+    TXT: {},
+    A: {}
+  },
+  types = {
+    1: 'A',
+    16: 'TXT'
+  },
+  apiServer;
+
+exports.serve = function(certDir, dnsPort, apiPort) {
+  dnsServer.on('request', function (request, response) {
+    var type = types[request.question[0].type];
+    var host = request.question[0].name.toLowerCase();
+    if (records[type] &&
+        records[type][host]) {
+      response.answer.push(dns[type]({
+        name: host,
+        address: records[type][host], //for A records
+        data: [records[type][host]], //for TXT records
+        ttl: 600,
+      }));
+    }
+    response.send();
+  });
+
+  dnsServer.on('error', function (err, buff, req, res) {
+    console.log(err.stack);
+  });
+
+  dnsServer.serve(dnsPort);
+
+  apiServer = https.createServer({
+    key: fs.readFileSync(certDir + '/privkey.pem'),
+    cert: fs.readFileSync(certDir + '/cert.pem'),
+    ca: fs.readFileSync(certDir + '/chain.pem'),
+    requestCert: true,
+    rejectUnauthorized: false
+  }, function(req, res) {
+    if (req.method !== 'POST') {
+      res.writeHead(405);
+      res.end('Please use POST method');
+      return;
+    }
+    try {
+      fingerprint = req.connection.getPeerCertificate().fingerprint.split(':').join('').toLowerCase();
+    } catch(e) {
+      res.writeHead(401);
+      res.end('Could not determine fingerprint of your client certificate');
+      return;
+    }
+    var expectedUrlParts = ['', 'v1', 'dns', 'net', 'useraddress', 'xob', fingerprint];
+    var urlParts = req.url.split('/');
+    function rejectUrlPath() {
+      res.writeHead(401);
```

Comment:
It's actually a 401 since it only allows you to edit under `<fingerprint>.<zoneRoot>` where `<fingerprint>` is the fingerprint of the client cert you present.